js/ui/grid_core/ui.grid_core.columns_controller.js

```diff
@@ -960,6 +960,20 @@ export const columnsControllerModule = {
                 return column;
             };
 
+            const sortColumns = (columns, sortOrder) => {
+                if(sortOrder !== 'asc' && sortOrder !== 'desc') {
+                    return columns;
+                }
+
+                const sign = sortOrder === 'asc' ? 1 : -1;
+
+                columns.sort(function(column1, column2) {
+                    return sign * column1.caption.localeCompare(column2.caption);
```

Comment:
What is happened if column.caption will be undefined? it can be for columns without dataField.